data_reduction/emu_yt_module.py

```diff
@@ -19,12 +38,29 @@ def __init__(self, time, kx, ky, kz, FT_magnitude, FT_phase):
         self.phase = FT_phase
 
 class EmuDataset(object):
-    def __init__(self, filename):
-        self.ds = yt.load(filename)
-        
+    def __init__(self, plotfile=None):
+        # initialize from a supplied plotfile, otherwise just make an empty dataset object
+        # and initialize it later with init_from_data()
+        if plotfile:
+            self.setup_dataset(yt.load(plotfile))
+        else:
+            self.ds = None
+
+    def init_from_data(self, data, left_edge=None, right_edge=None, dimensions=None,
+                       length_unit=(1.0, "cm"), periodicity=(True, True, True), nprocs=1):
+        # initialize the dataset using a dictionary of numpy arrays in data, keyed by field name
+        domain_bounds = np.array([[left_edge[0], right_edge[0]],
+                                  [left_edge[1], right_edge[1]],
+                                  [left_edge[2], right_edge[2]]])
+
+        self.setup_dataset(yt.load_uniform_grid(data, dimensions, length_unit=length_unit,
```

Comment:
Currently I don't think the time snapshot information is getting loaded with `.to_2D` or `.to_3D`. If we want this information to carry over, I think we need to add the `sim_time` parameter into the `yt.load_uniform_grid`. Otherwise it just says current_time = 0, which may show up in the volume renderings (in the current rendering script I have it display the time snapshot information).

Reply:
just pushed a fix, hopefully it works!